File: 7. Reverse Integer/main.cpp
#include <iostream>
#include <cstdint> // For int16_t

using namespace std;

int main()
{
    int32_t num = 4351214; // signed 32 bits int
    int32_t revNum = 0;
    int sign = num < 0 ? -1 : 1;
    num = abs(num);

    while (num != 0)
    {
        int digit = num % 10;
        if (num > (INT32_MAX - digit) / 10)
        {
            cout << "Invalid number";
            return 0;
        }
        revNum = (revNum * 10) + digit;
        num /= 10;
    }

    cout << revNum * sign;
}

File: 7. Reverse Integer/main.rs
fn main() {
    let mut num: i32 = 12238;
    let _sign: i8 = if num < 0 { -1 } else { 1 };
    let mut reverse_num: i32 = 0;
    num = num.abs();

    while num != 0 {
        let digit = num % 10;
        if num > ((i32::MAX - digit) / 10) {
            println!("THE END");
            break;
        }
        reverse_num = (reverse_num * 10) + digit;
        num /= 10;
    }

    println!("reverse ->{:?}", reverse_num * _sign as i32);
}
